### ISSUE TITLE
Version 0.4.0 crashes with "invalid memory address"

### ISSUE THREAD

**Issue by bitcoinlizard** [lightninglabs/lightning-terminal#177]
Version 0.3.4 ran without issue however version 0.4.0 is having a nil pointer error.  I experience this issue with the provided binary or building from source.  

I'm running go version 1.13.6.  I am using the linux-armv7 architecture.

Here is the error text:

2021-01-31 23:43:15.362 [INF] LITD: Handling gRPC web request: /poolrpc.Trader/NodeRatings
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x84be3c]

goroutine 555 [running]:
github.com/lightningnetwork/lnd/macaroons.(*Service).ValidateMacaroon(0x0, 0x1604150, 0x4302c00, 0x400f310, 0x1, 0x1, 0xf57278, 0x1b, 0xe7ba28, 0x42ef614)
        /home/#######/gocode/pkg/mod/github.com/lightningnetwork/lnd@v0.12.0-beta/macaroons/service.go:271 +0x198
github.com/lightninglabs/pool.(*Server).ValidateMacaroon(...)
        /home/#######/gocode/pkg/mod/github.com/lightninglabs/pool@v0.4.3-alpha/server.go:359
github.com/lightninglabs/lightning-terminal.(*LightningTerminal).ValidateMacaroon(0x41d1cc0, 0x1604150, 0x4302c00, 0x400f310, 0x1, 0x1, 0xf57278, 0x1b, 0x0, 0x5f3a100)
        /home/#######/lightning-terminal-source-v0.4.0-alpha/terminal.go:412 +0x248
github.com/lightninglabs/lightning-terminal.(*rpcProxy).UnaryServerInterceptor.func1(0x1604150, 0x4302ac0, 0xec6190, 0x4302ae0, 0x5f425a0, 0x5f425b0, 0x75c630, 0xe7ba28, 0x4302ac0, 0xe75610)
        /home/#######/lightning-terminal-source-v0.4.0-alpha/rpc_proxy.go:233 +0x170
github.com/lightninglabs/pool/poolrpc._Trader_NodeRatings_Handler(0xf23ef8, 0x4442de0, 0x1604150, 0x4302ac0, 0x54638f0, 0x41659d0, 0x1604150, 0x4302ac0, 0x5f3a180, 0x23)
        /home/#######/gocode/pkg/mod/github.com/lightninglabs/pool@v0.4.3-alpha/poolrpc/trader.pb.go:4144 +0x108
google.golang.org/grpc.(*Server).processUnaryRPC(0x4172500, 0x160d6b0, 0x44a4a00, 0x4495950, 0x42ccbe0, 0x1fc0be8, 0x0, 0x0, 0x0)
        /home/#######/gocode/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:1082 +0x3f8
google.golang.org/grpc.(*Server).handleStream(0x4172500, 0x160d6b0, 0x44a4a00, 0x4495950, 0x0)
        /home/#######/gocode/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:1405 +0xaa0
google.golang.org/grpc.(*Server).serveStreams.func1.1(0x4476c00, 0x4172500, 0x160d6b0, 0x44a4a00, 0x4495950)
        /home/#######/gocode/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:746 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/#######/gocode/pkg/mod/google.golang.org/grpc@v1.29.1/server.go:744 +0x7c

**Comment by guggero** [comment]
A few questions to narrow this down:
 - are you running remote mode or integrated mode?
 - is your `lnd` unlocked and fully started when you try to access the GUI?

**Comment by bitcoinlizard** [comment]
I installed LITD separately but used my existing binaries for LND (0.12.0), LOOP, and POOL.  Pool must be disabled for LITD to start as it must contain it's own pool daemon.  I am running LND and LOOP through systemd.  However LITD is running on the same machine.

Yes, LND is unlocked and running prior to launching LITD.  

One bit of extra info.  LITD did execute the demo educational walk through on first startup.  As soon as I completed the demo and switched to regular operating mode LITD crashed.
